ext/intl/converter/converter.stub.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+class UConverter
+{
+    public function __construct(?string $destination_encoding = null, ?string $source_encoding = null) {}
+
+    /** @return string|false */
+    public function convert(string $str, bool $reverse = false) {}
+
+    /**
+     * @param mixed $source
+     * @param mixed $codePoint
+     * @return string
+     */
+    public function fromUCallback(int $reason, $source, $codePoint, &$error) {}
+
+    /** @return array|false|null */
+    public static function getAliases(string $name) {}
+
+    /** @return array */
+    public static function getAvailable() {}
+
+    /** @return string|false|null */
+    public function getDestinationEncoding() {}
+
+    /** @return int|false|null */
+    public function getDestinationType() {}
+
+    /** @return int */
+    public function getErrorCode() {}
+
+    /** @return string|null */
+    public function getErrorMessage() {}
+
+    /** @return string|false|null */
+    public function getSourceEncoding() {}
+
+    /** @return int|false|null */
+    public function getSourceType() {}
+
+    /** @return array|null */
+    public static function getStandards() {}
+
+    /** @return string|false|null */
+    public function getSubstChars() {}
+
+    /** @return string|false */
+    public static function reasonText(int $reason) {}
+
+    /** @return bool */
+    public function setDestinationEncoding(string $encoding) {}
+
+    /** @return bool */
+    public function setSourceEncoding(string $encoding) {}
+
+    /** @return bool */
+    public function setSubstChars(string $chars) {}
+
+    /**
+     * @param mixed $source
+     * @param mixed $codePoint
+     * @return string|null
+     */
+    public function toUCallback(int $reason, $source, $codeUnits, &$error) {}
```

Comment:
Took me a while to understand WTF is going on here. Apparently this method exists so the class can be extended. I suspect that this code is incredibly inefficient for the case where you don't...

In any case, I think it should be fine if you use string parameters for source and codeUnits here and switch the zpp call accordingly.

Reply:
And what about the fromUCallback? Can I do the same with it? I don't even understand how these 2 parameters are used :O

Reply:
For fromUCallback it's apparently `int $reason, array $codeUnits, int $codePoint, &$error)`.